### PR TITLE
Driver:Serial: Add pin inversion to Esp32 Uart

### PR DESCRIPTION
**Pull request by AyushKot96**
### DESCRIPTION
 In this PR, additional properties are added to esp32 uart to allow signal inversion.

### CODE REVIEW

File: dts/bindings/serial/espressif,esp32-uart.yaml [hunk 1]
```diff
@@ -21,3 +21,17 @@ properties:
       Overrides hw-flow-control if both are set.
       Using this mode, the pin assigned to DTR
       is asserted during transmission.
+
+  tx-invert:
+    type: boolean
+    required: false
+    description: |
+      Invert the binary logic of tx pin. When enbaled, physical logic levels are inverted and
```
[the remark this answers is not in the document]
Comment (AyushKot96) [reply dts/bindings/serial/espressif,esp32-uart.yaml]: Fixed.

**Comment by sylvioalves** [comment]
@AyushKot96 Thanks for bringing this feature. Would you squash commits in a single one? Thanks.

File: dts/bindings/serial/espressif,esp32-uart.yaml [hunk 2]
```diff
@@ -21,3 +21,18 @@ properties:
...
+      Invert the binary logic of tx pin. When enabled, physical logic levels are inverted and 
+      we use 1=Low, 0=High instead of 1=High, 0=Low.
+
+  rx-invert:
+    type: boolean
+    required: false
+    description: |
+      Invert the binary logic of rx pin. When enabled, physical logic levels are inverted and 
+      we use 1=Low, 0=High instead of 1=High, 0=Low.
+
```
Comment (sylvioalves) [inline dts/bindings/serial/espressif,esp32-uart.yaml:38]: CI will fail due to missing blank line at the end here.

File: drivers/serial/uart_esp32.c [hunk 3]
```diff
@@ -328,6 +330,13 @@ static int uart_esp32_configure(const struct device *dev, const struct uart_conf
 
 	uart_hal_set_rx_timeout(&data->hal, 0x16);
 
+	if(config->tx_invert){
```
Comment (sylvioalves) [inline drivers/serial/uart_esp32.c:333]: I think these 2 ' if' cases are missing code style.. that will also fail. Please, fix it.

File: dts/bindings/serial/espressif,esp32-uart.yaml [hunk 4]
```diff
@@ -21,3 +21,16 @@ properties:
...
+    description: |
+      Invert the binary logic of tx pin. When enabled, physical logic levels are inverted and
+      we use 1=Low, 0=High instead of 1=High, 0=Low.
+
+  rx-invert:
+    type: boolean
+    description: |
+      Invert the binary logic of rx pin. When enabled, physical logic levels are inverted and
+      we use 1=Low, 0=High instead of 1=High, 0=Low.
+
```
Comment (sylvioalves) [inline dts/bindings/serial/espressif,esp32-uart.yaml:36]: Sorry, in last commit you missed a line break, not a blank line.

**Comment by sylvioalves** [comment]
Break down the commit description in 2 lines to meet compliance rules as it exceeds 75 chars.

**Comment by AyushKot96** [comment]
> Break down the commit description in 2 lines to meet compliance rules as it exceeds 75 chars.

Thank you for bearing with me. This is the first time I am creating a pull request in the open source community. 😃

**Review by LucasTambor — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

File: dts/bindings/serial/espressif,esp32-uart.yaml [hunk 5]
```diff
@@ -21,3 +21,15 @@ properties:
...
       Using this mode, the pin assigned to DTR
       is asserted during transmission.
+
+  tx-invert:
+    type: boolean
+    description: |
+      Invert the binary logic of tx pin. When enabled, physical logic levels are inverted and
+      we use 1=Low, 0=High instead of 1=High, 0=Low.
+
+  rx-invert:
```
Comment (decsny) [inline dts/bindings/serial/espressif,esp32-uart.yaml:31]: these same properties were just added on an NXP controller in https://github.com/zephyrproject-rtos/zephyr/pull/71802/

should we make a generic property?

**Review by ubieda — APPROVED** [review]

File: dts/bindings/serial/espressif,esp32-uart.yaml (hunk 5 above)
Comment (AyushKot96) [reply dts/bindings/serial/espressif,esp32-uart.yaml:31]: Yes, I think we should make a generic property in uart driver in the future.

Comment (marekmatej) [reply dts/bindings/serial/espressif,esp32-uart.yaml:31]: why not updating the `uart-controller.yaml` now?

Comment (cx-anuj-pathak) [reply dts/bindings/serial/espressif,esp32-uart.yaml:31]: Here is the PR https://github.com/zephyrproject-rtos/zephyr/pull/72947

**Pull request merged**